src/ajax/xhr.js

```diff
@@ -88,12 +88,15 @@ jQuery.ajaxTransport( function( options ) {
 									xhrSuccessStatus[ xhr.status ] || xhr.status,
 									xhr.statusText,
 
-									// Support: IE9
-									// Accessing binary-data responseText throws an exception
-									// (#11426)
-									typeof xhr.responseText === "string" ? {
-										text: xhr.responseText
-									} : undefined,
+									// XHR2 responseText throws an exception on binary data,
+									// return it raw and let a converter or caller handle it
+									// https://xhr.spec.whatwg.org/#the-responsetext-attribute
+									// (trac-11426, gh-2498)
+									xhr.responseType === "arraybuffer" ||
```

Comment:
> once we drop IE9 support after 3.0

You're really forward-thinking! ]:->

FWIW, even Angular 2 will support even IE9 in the end (Amadeus that works on Angular with Google has to support IE9 because of its high market share in some countries like South Korea) so I don't envision us dropping support soon.